aioshelly/rpc_device/device.py

```diff
@@ -252,10 +255,19 @@ async def trigger_reboot(self, delay_ms: int = 1000) -> None:
     async def update_status(self) -> None:
         """Get device status from 'Shelly.GetStatus'."""
         self._status = await self.call_rpc("Shelly.GetStatus")
+        self._status.update(
+            {
+                item["key"]: {"value": item["status"].get("value")}
```

Comment:
`button` virtual component has no `value` in the `status` so we need use `get()` here.

Reply:
I think it would be cleaner to move this to line 193 (after we fetch status) and make `get_dynamic_components` update the status and the config right after fetching the data

Reply:
Just notice we fetch status at
https://github.com/home-assistant/core/blob/ac8ea8356c8483c2123ba9c12c3cd1b7652093ab/homeassistant/components/shelly/coordinator.py#L741

So if we do the change I suggested we will need to call `get_dynamic_components` there so maybe also